File: iosApp/iosApp/Design/KdSpacing.swift
import CoreGraphics

// ---------------------------------------------------------------------------
// KdSpacing — 4-pt grid scale (SwiftUI translation of compose-ui KdSpacing.kt)
// Values in points (1pt = 1dp on iOS at 1x scale).
// ---------------------------------------------------------------------------

enum KdSpacing {
    static let s1: CGFloat = 4
    static let s2: CGFloat = 8
    static let s3: CGFloat = 12
    static let s4: CGFloat = 16
    static let s5: CGFloat = 20
    static let s6: CGFloat = 24
    static let s7: CGFloat = 32
    static let s8: CGFloat = 40
    static let s9: CGFloat = 48
    /// S03 inter-element gap
    static let gap: CGFloat = 14
    /// C01 84-pt avatar (chat empty state, single-device pair)
    static let heroAvatar: CGFloat = 84
}

File: compose-ui/src/commonMain/kotlin/com/carlom/klardrop/theme/KdSpacing.kt
package com.carlom.klardrop.theme

import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp

// ---------------------------------------------------------------------------
// KdSpacing — 4-pt grid scale from spec/tokens.css plus a few spec-named sizes
// that aren't on the 4-pt grid (gap, heroAvatar, sheet) but are mandated by
// the implementation spec.
// ---------------------------------------------------------------------------

data class KdSpacing(
    val s1: Dp = 4.dp,
    val s2: Dp = 8.dp,
    val s3: Dp = 12.dp,
    val s4: Dp = 16.dp,
    val s5: Dp = 20.dp,
    val s6: Dp = 24.dp,
    val s7: Dp = 32.dp,
    val s8: Dp = 40.dp,
    val s9: Dp = 48.dp,
    val gap: Dp = 14.dp,         // S03 inter-element gap
    val heroAvatar: Dp = 84.dp,  // C01 84-dp avatar (chat empty, single-device pair)
)

val KdDefaultSpacing = KdSpacing()
